### DISCUSSION TITLE
JSON Pointer support in On Demand

### DISCUSSION THREAD

**Discussion by erichutchins** [simdjson/simdjson#1398] [General]
Hi all, thank you for this amazing library. Do you plan to support json pointers in the on demand api? Since I saw the change to support `tweets["search_metadata"]["count"]` syntax, I’ve been hopeful json pointers could also be possible. I haven’t seen any discussion of this in the issues.

This is a very common use case for me: `$ zstdcat logs.ndjson.zst | jq “.extract.deeply.nested[3].key”`

I’ve successfully used the dom api with a basic getline to speed this up significantly. Streaming ndjson with on demand json pointer retrieval would be awesome.

**Comment by jkeiser** [comment]
It's definitely in the cards, though we hadn't made a tracking issue, thanks!

The first priority for on demand has been getting it solid, safe and fast, so we can build things like json pointer on top of it.

Contributions are welcome, too!

**Comment by lemire** [comment]
> we hadn't made a tracking issue 

It is not yet an issue.

> The first priority for on demand has been getting it solid, safe and fast, so we can build things like json pointer on top of it.

Can runtime pointers be implemented on top of on demand, given that on demand is a compile-time approach?

**Comment by jkeiser** [comment]
I think it can, along with recursion and generic type walking. I suspect it won't lose too much in speed, but time will tell. Either way, I think it's appealing to be able to use json pointer to find something and then use the rest of on demand to process it.
